File: oolang-stdlib/src/main/java/oo/internal/Expected.java
/*
 * This is free and unencumbered software released into the public domain, following <https://unlicense.org>
 */

package oo.internal;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface Expected {
	/**
	 * Returns the modifier.
	 * @return the modifier
	 */
	Modifier value();
}

File: oolang-stdlib/src/main/kotlin/oo/internal/Expected.kt
/*
 * This is free and unencumbered software released into the public domain, following <https://unlicense.org>
 */

package oo.internal

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
annotation class Expected(
        val modifier: Modifier
)
